indexer/data_source.hpp

```diff
@@ -0,0 +1,314 @@
+#pragma once
+
+#include "indexer/cell_id.hpp"
+#include "indexer/feature.hpp"
+#include "indexer/feature_covering.hpp"
+#include "indexer/features_offsets_table.hpp"
+#include "indexer/feature_source.hpp"
+#include "indexer/features_vector.hpp"
+#include "indexer/mwm_set.hpp"
+#include "indexer/scale_index.hpp"
+#include "indexer/unique_index.hpp"
+
+#include "coding/file_container.hpp"
+
+#include "base/macros.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "defines.hpp"
+
+template<typename FeatureSource, typename Element>
+class ReadMWMFunctor final
+{
+  using Fn = std::function<void(Element &)>;
+
+public:
+  explicit ReadMWMFunctor(Fn const & fn) : m_fn(fn) {}
+
+  template<class T>
+  typename std::enable_if<std::is_same<T, FeatureID const>::value, void>::type
```

Comment:
1. Please use enable_if_t
2. Are you sure you need enable_if here (instead of renaming to ProcessFeatureID, say)?

Reply:
1. done
2. sure